MetaMask/metamask-extension
# Toggle option to enable/disable balance and Token rate checking for using third-party API

Fixes #16724 
## Explanation
We have added a new toggle to the `Privacy and Security` tab to control the use of third-party APIs like coinGecko and CryptoCompare for balance and Token rate checking.


## Screenshots/Screencaps



### Before
<img width="1022" alt="Screenshot 2022-12-07 at 6 00 34 PM" src="https://user-images.githubusercontent.com/43930900/206315461-0e2bb329-54e9-40f5-8019-69819be6a2e2.png">
<img width="324" alt="Screenshot 2022-12-07 at 6 00 57 PM" src="https://user-images.githubusercontent.com/43930900/206315526-284c151f-1025-484f-baab-c92b0a06b416.png">
<img width="411" alt="Screenshot 2022-12-07 at 6 01 47 PM" src="https://user-images.githubusercontent.com/43930900/206315655-e117c724-b7c8-411a-b5cc-0e453b195629.png">
<img width="409" alt="Screenshot 2022-12-07 at 6 02 09 PM" src="https://user-images.githubusercontent.com/43930900/206315713-b71dfee5-5392-4488-8f79-281e27653476.png">



### After
<img width="1021" alt="Screenshot 2022-12-07 at 5 59 02 PM" src="https://user-images.githubusercontent.com/43930900/206315226-eebac9e9-3cef-43ad-8fd1-667ae65b8cb9.png">
<img width="326" alt="Screenshot 2022-12-07 at 5 58 28 PM" src="https://user-images.githubusercontent.com/43930900/206315114-22e9d9b8-82e3-4983-9a1b-be57ad718d90.png">
<img width="417" alt="Screenshot 2022-12-07 at 5 59 31 PM" src="https://user-images.githubusercontent.com/43930900/206315310-d3b07342-b9e4-40cf-8911-dc6926671af3.png">
<img width="406" alt="Screenshot 2022-12-07 at 5 59 52 PM" src="https://user-images.githubusercontent.com/43930900/206315363-718d1a6c-8a16-4592-be6d-7d5e84f26126.png">




## Manual Testing Steps
Onboard v1:
  1. Load the application and import a wallet.
  2. The fiat values should not be visible - in eth-overview, token overview, or in asset/token lists
  3. Try to send a token, and fiat values for gas or total should not be visible.
  4. Now go to settings, search for `Show balance and token price checker` in the security tab and toggle it ON.
  5. Now, come back to the overview page and make sure Fiat values are visible.
  6. Again, Toggle OFF and make sure the Fiat values disappear from everywhere.
  
Onboarding V2:
1. Load the application and import a wallet.
2. Enable the `Show balance and token price checker` from the Privacy settings.
3. The app will display the fiat values.
4. reload the application and import the wallet again.
5. This time, without enabling `Show balance and token price checker`.
6. the Fiat values should not e displayed.


## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
